sdk/storage/azure-storage-queue/tests/test_queue.py

```diff
@@ -405,6 +383,39 @@ def test_get_messages_with_options(self, storage_account_name, storage_account_k
             self.assertNotEqual('', message.expires_on)
             self.assertNotEqual('', message.next_visible_on)
 
+    @QueuePreparer()
+    def test_get_messages_with_max_messages(self, storage_account_name, storage_account_key):
```

Comment:
Could we please add another test of the reverse scenario?
I.e. load a queue up with 5 messages, and set max_messages=10

Reply:
We should also test other combinations of parameters - for example, if I set `messages_per_page=3`, then I set `max_messages=10`, we should confirm that our logic isn't actually gathering 12 messages and discarding the last 2 (the current implementation shouldn't have this problem, but by testing for it, we can protect against future regressions).